### PR TITLE
Fix item info pos and height

### LINKED ISSUE #98266: rustdoc: item-infos have different vertical size
On pages that have an item-info for "experimental" and also a platform restriction (e.g. https://doc.rust-lang.org/nightly/std/arch/macro.is_arm_feature_detected.html), the two item-infos are different sizes:

image

The platform restriction should be the same height as the experimental item-info.

### LINKED ISSUE #98343: Deprecation note and doc(cfg) don't look good in combination
I think there should be some spacing in between, and the different height is also unfortunate.

Screenshot_2022-06-21_155629

I also opened #98341 about the poor contrast of the link, but that's unrelated to `doc(cfg)`.

@rustbot labels: +T-rustdoc +F-doc_cfg +A-rustdoc-ui

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rust-lang/rust/issues/98266.
Fixes https://github.com/rust-lang/rust/issues/98343.

You can test it [here](https://rustdoc.crud.net/imperio/fix-item-info-pos-and-height/lib2/trait.Trait.html).

Here is a screenshot of the result:

Screenshot from 2022-07-26 21-55-53

r? @jsha

### CODE REVIEW

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -1111,7 +1111,14 @@ table,
 }
 
 .item-info .stab {
-	display: inline-block;
+	width: fit-content;
+	/* This min-height is needed to unify the height of the stab elements because some of them
+	   have emojis.
+	*/
```
Comment (camsteffen) [inline src/librustdoc/html/static/css/rustdoc.css:1117]: Idea:

```css
.stab::before {
  content: '\200b'; /* zero-width space */
  font-size: 1.25rem; /* emoji font size */
}
```

Maybe the `content` could be the actual emoji when applicable, determined by a CSS class?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1117]: The `min-height` is mostly for items which don't have an emoji actually. Unless I misunderstood what you meant?

Comment (camsteffen) [reply src/librustdoc/html/static/css/rustdoc.css:1117]: Right. The zero-with space makes the element taller as if there were an emoji.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1117]: Exactly.

Comment (camsteffen) [reply src/librustdoc/html/static/css/rustdoc.css:1117]: So you wouldn't need an arbitrary min-height.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1117]: Sorry, I still don't get your point.

Comment (camsteffen) [reply src/librustdoc/html/static/css/rustdoc.css:1117]: Sorry, I guess this is all moot. I thought that the actual height of the emoji is not a predictable px value. I've had experiences with some fonts not having a predictable height.

**Pull request merged**